### DISCUSSION TITLE
AvatarBuilder to be of type `Function(User)?` instead of `Function(String)?`

### DISCUSSION THREAD

**Discussion by allenkaplan** [flyerhq/flutter_chat_ui#286] [Ideas & Requests]
Currently the custom avatar builder introduced by #248 has the AvatarBuilder take in the userId (type String)

I would like to have it take in the User so I can access `imageUrl` in addition to `userId` 

The rationale is that we want the AvatarBuilder to use the CachedNetworkImage library (https://pub.dev/packages/cached_network_image) with a `cacheKey` of `userId`, but the still have the url to download in case its missing in the cache for some reason

If this is has support from maintainers/community, I'm happy to change this so we can use this in our own implementation

**Comment by demchenkoalex** [comment]
As far as I remember (@felixgabler correct me if I'm wrong) the rationale behind that MR was that the user can be updated elsewhere, and the builder would return user ID for you to fetch user first (to have the latest data) and then get `imageUrl` from there. Is that correct? If we would return user it will be the old user, not necessarily having the actual image.

**Reply by felixgabler** [reply]
Actually, if we change the function as suggested, we would not lose any functionality but only extend the possibilities. Back then, I assumed that people just need the ID to fetch data and not the whole user. But your requirements show new needs that we should resolve. So I'd say go ahead :)

**Comment by demchenkoalex** [comment]
@AllenKaplan please open a PR if you may, thanks!
